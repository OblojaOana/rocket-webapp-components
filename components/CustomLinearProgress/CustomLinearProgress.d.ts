import { LinearProgressProps } from "@material-ui/core";

export interface CustomLinearProgressProps extends Omit<LinearProgressProps, 'color' | 'variant' | 'value'> {
        /**
        *  The color of the component.
        * @default "gray"
        */
        color?: "primary" |
        "warning" |
        "danger" |
        "success" |
        "info" |
        "rose" |
        "gray"
        /**
        * The variant to use. Use indeterminate or query when there is no progress value.
        * @default "indeterminate"
        */
        variant?: "determinate" |
        "indeterminate" |
        "buffer" |
        "query"
        /**
        * The value of the progress indicator for the determinate and buffer variants. Value between 0 and 100.
        */
        value?: number
}
/**
 *
 * Demos:
 *
 * - https://bit.dev/totalsoft_oss/react-mui/custom-linear-progress
 *
 *  Take a look over the code for further information:https://github.com/osstotalsoft/rocket-webapp-components/blob/develop/components/CustomLinearProgress/CustomLinearProgress.js
 */
export default function CustomLinearProgress(props: CustomLinearProgressProps): JSX.Element;