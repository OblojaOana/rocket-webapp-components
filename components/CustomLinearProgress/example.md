##### Copy this snapshot in the overview panel
 ```js
 import React from 'react';
import CustomLinearProgress from '@bit/totalsoft_oss.react-mui.custom-linear-progress';

export default (
	<CustomLinearProgress
		color="primary"
		value={20}
		variant="determinate"
	></CustomLinearProgress>
)
 ```