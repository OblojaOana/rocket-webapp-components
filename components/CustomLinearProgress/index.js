import CustomLinearProgress from "../../core/CustomLinearProgress";
export default CustomLinearProgress;