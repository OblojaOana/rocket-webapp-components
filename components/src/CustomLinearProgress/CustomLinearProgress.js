import React from "react";
import PropTypes from "prop-types";
import { makeStyles, LinearProgress } from "@material-ui/core";
import customLinearProgressStyle from "./customLinearProgressStyle";

const useStyles = makeStyles(customLinearProgressStyle);

export default function CustomLinearProgress(props) {
    const { color, variant, value, ...rest } = props;
    const classes = useStyles();
    return (
        <LinearProgress
            value={value} variant={variant} {...rest}
            classes={{
                root: classes.root + " " + classes[color + "Background"],
                bar: classes.bar + " " + classes[color]
            }}
        />
    );
}

CustomLinearProgress.defaultProps = {
    color: "gray",
    variant: "indeterminate"
};

CustomLinearProgress.propTypes = {
    color: PropTypes.oneOf([
        "primary",
        "warning",
        "danger",
        "success",
        "info",
        "rose",
        "gray"
    ]),
    variant: PropTypes.oneOf([
        "determinate",
        "indeterminate",
        "buffer",
        "query"
    ]),
    value: PropTypes.number
};