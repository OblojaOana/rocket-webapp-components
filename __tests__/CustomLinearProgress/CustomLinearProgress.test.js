import React from 'react';
import renderer from 'react-test-renderer';
import CustomLinearProgress from '../../components/CustomLinearProgress/CustomLinearProgress';

describe('CustomLinearProgress', () => {
    it('snapshot', () => {
        const component = renderer.create(<CustomLinearProgress
            color="primary"
            value={20}
            variant="determinate"
        />);

        const json = component.toJSON();
        expect(json).toMatchSnapshot();
    });
});