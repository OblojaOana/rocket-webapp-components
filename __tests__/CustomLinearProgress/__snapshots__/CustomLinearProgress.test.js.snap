// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomLinearProgress snapshot 1`] = `
<div
  aria-valuemax={100}
  aria-valuemin={0}
  aria-valuenow={20}
  className="MuiLinearProgress-root makeStyles-root-1 makeStyles-primaryBackground-10 MuiLinearProgress-colorPrimary MuiLinearProgress-determinate"
  role="progressbar"
>
  <div
    className="MuiLinearProgress-bar makeStyles-bar-2 makeStyles-primary-3 MuiLinearProgress-barColorPrimary MuiLinearProgress-bar1Determinate"
    style={
      Object {
        "transform": "translateX(-80%)",
      }
    }
  />
</div>
`;
